crates/cli/src/introspection/type_unification.rs

```diff
@@ -17,8 +17,8 @@ type ObjectType = WithName<schema::ObjectType>;
 /// Unify two types.
 /// This is computing the join (or least upper bound) of the two types in a lattice
 /// where `ExtendedJSON` is the Top element, Scalar(Undefined) is the Bottom element, and Nullable(T) >= T for all T.
-pub fn unify_type(type_a: Type, type_b: Type) -> Type {
-    let result_type = match (type_a, type_b) {
+pub fn unify_type(type_a: Type, type_b: Type, all_schema_nullable: bool) -> Type {
```

Comment:
`unify_type` should not need to know about `all_schema_nullable` option. This should be handled in `make_field_type` by wrapping the resulting type in `Nullable`.